usagov/usagov-2021
# Usagov 1997 mobile menu

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1997

## Description

Moves generation of the mobilemenu block out of theme twig files and into usagov_menus module
Fixes an infrequent, race-condition bug where the mobile menu on agency pages was not correctly built when exported by tome.

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



### Change Requirements



### Validation Steps

- Make two exports for comparison
    - make a tome export of local using the dev branch, rename html/ to html-dev/ Also in the usagov_ssh_postprocessing module, renamed save the published pages CSV for the dev run to compare in the next step.
    -  switch to this branch, make a tome export, rename to html-dev
- Compare HTML(html-tidy config attached)
    - use html-tidy + find command to "normalize" the HTML output in each directory. Use the option to remove indentation. In each HTML folder run a command like:
    `find . -name '*.html' -type f -print -exec tidy -config ../tidy-config.txt --warn-proprietary-attributes false -mq '{}' \;`
    - use diff + grep to compare the contents of both directories to confirm the HTML is same between branches. Something like this where diff ignores the cache busting query string drupls adds to CSS and JS files will give you a file to review in a text editor..:
    `diff -brwd --strip-trailing-cr --exclude=_data -I " rel=\"stylesheet\">" -I "<script src=" -I js-view-dom-id -I mobilemenublock --exclude=themes html-dev/ html/ > html.diff `
    - The diff file above should not have any lines starting with angle brackets. If you're dev export didn't have the correct agency mobile menu on some pages, you should see the correct one in this branch's export.
- Compare number of redirects in each export directory to confirm its the same. Use this command inside each directory:
`grep -r 'http-equiv=\"refresh\"' . | wc -l`
 



## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
